static/posts/how-to-build-a-blog.md

```diff
@@ -0,0 +1,258 @@
+[handler](https://fresh.deno.dev/docs/getting-started/custom-handlers) to
+[fetch](https://fresh.deno.dev/docs/getting-started/fetching-data) data before
+rendering a page.
+
+Following the examples in the docs, let's go ahead and create a handler for our
+page and use it to pass data to the page component:
+
+```tsx
+// routes/posts/[name].tsx
+
+import { Handlers, PageProps } from "$fresh/server.ts";
+import Markdown from "../../components/Markdown.tsx";
+
+// A map of post paths to Markdown. We'll leave it empty for now
+const postMap = new Map<string, string>();
+
+export const handler: Handlers<string> = {
+  GET: (_, ctx) => {
+    const { name } = ctx.params;
+    // Check if the post exists
+    const post = postMap.get(name);
+    if (post === undefined) { // Render a 404 page if it doesn't
+      return ctx.renderNotFound();
+    }
+    return ctx.render(post); // Render a post page if it does
+  },
+};
+
+// Let's modidfy our render function to use the Markdown component with
+// the Markdown passed from our handler.
+export default (props: PageProps<string>) => (
+  <>
+    <Markdown markdown={props.data} />
+  </>
+);
+```
+
+**TODO**
+
+- You will probably want to add more to this page. See the
+  [Post Frontmatter](#post-frontmatter) section below to learn about adding
+  things like a title, date, or tags to each post.
+- Fresh provides default error pages, but if you would like to customize them be
+  sure to checkout the docs
+  [here](https://fresh.deno.dev/docs/concepts/error-pages).
+
+### Loading Posts from a Directory
+
+Next, we need to load the posts we've written into `postMap`. Let's use Deno to
+read our Markdown files from a directory. I would also recommend creating a new
+file to hold this code so that `routes/posts/[name].tsx` doesn't get too
+cluttered.
+
+```tsx
+// utils/posts.ts
+
+import { basename, extname } from "$std/path/mod.ts";
+
+// Let's move postMap here
+export const postMap = new Map<string, string>();
+
+// Let's create a constant for the directory to our posts so that if it
+// ever needs to be changed, we can do it in one location.
+const POSTS_DIR_PATH = "./static/posts/";
+
+// Load the posts from the file system
+for await (const dirEntry of Deno.readDir(POSTS_DIR_PATH)) {
+  const { name: filename } = dirEntry;
+  if (dirEntry.isFile && filename.endsWith(".md")) {
+    const postPath = `${POSTS_DIR_PATH}${filename}`;
+    const fileContents = await Deno.readTextFile(postPath);
+    // We'll use the basename of each filename as the key in the postMap
+    const postBasename = basename(filename, extname(filename));
+    postMap.set(postBasename, fileContents);
+  }
+}
+```
+
+Now we have basically everything we need! I'd recommend creating some test posts
+to see how they look.
+
+### Post Frontmatter
+
+Although we have the ability to render posts, we don't have a great way to
+browse through posts or show information about each post in a uniform way. In
```

Comment:
"we don't have a great way to browse through **posts** or show information about each post in a uniform way."

Replace the bolded "posts" with "them" - pronouns help a lot with making sure repeating words don't mess up the rhythm of a sentence while reading.

Reply:
Realizing now that the boldness on the font isn't super noticeable. To be clear: it's the only instance of the word "posts" (with an 's' at the end) in the quoted section.